lib/interpreters/split_clustering.rb

```diff
@@ -793,13 +793,19 @@ def self.collect_cluster_data(vrp, nb_clusters)
     def self.duplicate_vehicle(vehicle, timewindow, schedule)
       if timewindow.nil?
         (schedule[:start]..[schedule[:end], 6].min).collect{ |day|
```

Comment:
The following code is less error prone and correct compared to the previous code and handles those "makes no sense" cases naturally without fuss
```suggestion
        (0..6).flat_map{ |day|
          next unless (schedule[:start]..schedule[:end]).any?{ |day_index| day_index % 7 == day }
```